openmc/source.py

```diff
@@ -212,36 +281,23 @@ def time(self, time):
         cv.check_type('time distribution', time, Univariate)
         self._time = time
 
-    @strength.setter
-    def strength(self, strength):
-        cv.check_type('source strength', strength, Real)
-        cv.check_greater_than('source strength', strength, 0.0, True)
-        self._strength = strength
-
     @particle.setter
     def particle(self, particle):
         cv.check_value('source particle', particle, ['neutron', 'photon'])
         self._particle = particle
 
-    def to_xml_element(self) -> ET.Element:
-        """Return XML representation of the source
+    def populate_xml_element(self, element):
+        """Add necessary source information to an XML element
 
         Returns
         -------
         element : lxml.etree._Element
             XML element containing source data
 
         """
-        element = ET.Element("source")
-        element.set("strength", str(self.strength))
+        super().populate_xml_element(element)
         if self.particle != 'neutron':
             element.set("particle", self.particle)
```

Comment:
I have no qualm with writing the particle type to XML regardless of type. I think this follows the zen of Python that we generally try to follow. Just wasn't a focus of this PR. Might as well make the update here since many of the test inputs are changing anyway unless others have thoughts on this @paulromano @gridley?

Reply:
Makes sense to me. Since OpenMC XML is de facto not written by humans these days, increasing both its explicitness and verbosity makes perfect sense.

Reply:
Yeah, that sounds fine to me!

Reply:
Done deal! Just added this in.